src/app/wallets/get-transactions-for-wallet.ts

```diff
@@ -1,14 +1,22 @@
-import { PartialResult } from "@app/partial-result"
+import { BTC_NETWORK, ONCHAIN_MIN_CONFIRMATIONS, SECS_PER_5_MINS } from "@config"
+
 import { getCurrentPrice } from "@app/prices"
-import { BTC_NETWORK, ONCHAIN_MIN_CONFIRMATIONS } from "@config"
-import { OnChainError, TxDecoder, TxFilter } from "@domain/bitcoin/onchain"
-import { RepositoryError } from "@domain/errors"
+import { PartialResult } from "@app/partial-result"
+
+import { CacheKeys } from "@domain/cache"
 import { LedgerError } from "@domain/ledger"
+import { toSeconds } from "@domain/primitives"
+import { RepositoryError } from "@domain/errors"
 import { WalletTransactionHistory } from "@domain/wallets"
-import { LedgerService } from "@services/ledger"
-import { OnChainService } from "@services/lnd/onchain-service"
+import { OnChainError, TxDecoder, TxFilter } from "@domain/bitcoin/onchain"
+
 import { baseLogger } from "@services/logger"
+import { LedgerService } from "@services/ledger"
+import { RedisCacheService } from "@services/cache"
 import { WalletsRepository } from "@services/mongoose"
+import { OnChainService } from "@services/lnd/onchain-service"
+
+const redisCache = RedisCacheService()
```

Comment:
what drive your decision to use Redis instead of a local cache?

because cache can be invalidated from trigger?

 do you have a honeycomb query to see how the latency distribution is from `onChain.listIncomingTransactions(ONCHAIN_MIN_CONFIRMATIONS)`?

Reply:
https://ui.honeycomb.io/galoy/datasets/galoy-bbw/result/4tkafh5ZGbg it is not only the latency but also the amount of requests to lnd because it is in the main query (specially when lnd has problems)

Reply:
this is also a good query to understand in what methods we should improve performance and amount of requests https://ui.honeycomb.io/galoy/datasets/galoy-bbw/result/oXTxLx6jKbK

Reply:
> this is also a good query to understand in what methods we should improve performance and amount of requests https://ui.honeycomb.io/galoy/datasets/galoy-bbw/result/EGqhArMX7R2

yup. lookup invoice would really benefits for using holdinvoices